diff -ur old/board/sdp/leonid/config.mk new/board/sdp/leonid/config.mk
--- old/board/sdp/leonid/config.mk	2008-06-30 03:35:20.000000000 +0200
+++ new/board/sdp/leonid/config.mk	2010-05-28 07:35:07.000000000 +0200
@@ -12,5 +12,5 @@
 #
 
 TEXT_BASE = 0x67c00000
-USE_TBML = y
+#USE_TBML = y
 #USE_BBML = y
diff -ur old/common/cmd_nvedit.c new/common/cmd_nvedit.c
--- old/common/cmd_nvedit.c	2008-06-29 10:53:01.000000000 +0200
+++ new/common/cmd_nvedit.c	2010-05-28 07:29:06.000000000 +0200
@@ -595,38 +595,6 @@
 	"    - delete environment variable 'name'\n"
 );
 
-#ifndef CFG_NO_FLASH
-#if defined(CFG_ENV_IS_IN_NVRAM) || defined(CFG_ENV_IS_IN_EEPROM) || \
-//    ((CONFIG_COMMANDS & (CFG_CMD_ENV|CFG_CMD_FLASH)) == \
-//      (CFG_CMD_ENV|CFG_CMD_FLASH)) || \
-    ((CONFIG_COMMANDS & (CFG_CMD_ENV|CFG_CMD_NAND)) == \
-      (CFG_CMD_ENV|CFG_CMD_NAND)) 
-U_BOOT_CMD(
-	saveenv, 1, 0,	do_saveenv,
-	"saveenv - save environment variables to persistent storage\n",
-	NULL
-);
-#endif
-#endif	/* CFG_CMD_ENV */
-
-#if   ((CONFIG_COMMANDS & (CFG_CMD_ENV|CFG_CMD_ONENAND)) == \
-      (CFG_CMD_ENV|CFG_CMD_ONENAND))
-int do_saveenv (cmd_tbl_t *cmdtp, int flag, int argc, char *argv[])
-{
-	extern char * env_name_spec;
-
-	printf ("Saving Environment to %s...\n", env_name_spec);
-
-	return (saveenv() ? 1 : 0);
-}
-      
-U_BOOT_CMD(
-	saveenv, 1, 0,	do_saveenv,
-	"saveenv - save environment variables to persistent storage\n",
-	NULL
-);
-#endif
-
 
 #if (CONFIG_COMMANDS & CFG_CMD_ASKENV)
 
diff -ur old/common/main.c new/common/main.c
--- old/common/main.c	2008-07-03 13:26:50.000000000 +0200
+++ new/common/main.c	2010-05-28 07:19:30.000000000 +0200
@@ -435,9 +435,6 @@
 	/* This point is never reached */
 	for (;;);
 #else
-	/* sh.kim - start */
-		samsung_prompt();
-	/* sh.kim - end */
 	for (;;) {
 #ifdef CONFIG_BOOT_RETRY_TIME
 		if (rc >= 0) {
diff -ur old/include/configs/leonid.h new/include/configs/leonid.h
--- old/include/configs/leonid.h	2010-05-28 07:36:27.000000000 +0200
+++ new/include/configs/leonid.h	2010-05-28 07:24:10.000000000 +0200
@@ -73,12 +73,10 @@
 			CFG_CMD_NET		| \
 			CFG_CMD_RUN		| \
 			CFG_CMD_PCI		| \
-			CFG_CMD_ONENAND		| \
 			CFG_CMD_USB		| \
 			CFG_CMD_FAT		| \
 			0ULL) & ~(0ULL		| \
 			CFG_CMD_IMLS		| \
-			CFG_CMD_FLASH		| \
 			0ULL)
 
 #include <cmd_confdefs.h>
@@ -111,7 +109,7 @@
 
 /* environment info location */
 #define CONFIG_ENV_OVERWRITE
-#define CFG_ENV_IS_IN_ONENAND
+#define CFG_ENV_IS_NOWHERE
 #define CFG_ENV_ADDR			0x80000
 #define CFG_ENV_SIZE			0x00020000
 #define CFG_ENV_SECT_SIZE		0x00020000
diff -ur old/lib_arm/board.c new/lib_arm/board.c
--- old/lib_arm/board.c	2008-12-08 10:33:01.000000000 +0100
+++ new/lib_arm/board.c	2010-05-28 07:17:51.000000000 +0200
@@ -200,7 +200,7 @@
 }
 #endif /* CFG_NO_FLASH */
 
-#if 1
+#if 0
 int env_setup(void)
 {
 	char *buffer1;
@@ -598,27 +598,10 @@
 	eth_initialize(gd->bd);
 #endif
 
-    *(volatile unsigned int*)0x30090c8c &= ~(0xf << 4); /* gpio5.1 : UART0_RX port */
-    *(volatile unsigned int*)0x30090c8c |= (0x2 << 4);  /* gpio5.1  set input */
-
-    upgrade_flag = (*(volatile unsigned int*)0x30090c94)>>1 & 0x1;
-    for(i=0; i<500000; i++);
-    upgrade_flag = (*(volatile unsigned int*)0x30090c94)>>1 & 0x1;
-
-    *(volatile unsigned int*)0x30090c8c &= ~(0xf << 4); /* gpio5.1 : UART0_RX port set main*/
-        
-    if(upgrade_flag == 0) /* check update_flag is not 0 for sw update*/
-    {
-        /* if update_flag is not 0 then sw update */
-        bbm_swupdate();
-    }
-    else
-    {
 	/* main_loop() can return to retry autoboot, if so just run it again. */
 	for (;;) {
 		main_loop ();
         	}
-    }
 
 	/* NOTREACHED - no way out of command loop except booting */
 }
