--- a/Makefile	2010-06-16 13:55:48.000000000 +0200
+++ b/Makefile	2010-06-16 14:08:19.000000000 +0200
@@ -191,8 +191,8 @@
 # Note: Some architectures assign CROSS_COMPILE in their arch/*/Makefile
 export KBUILD_BUILDHOST := $(SUBARCH)
 ARCH			:= arm
-CROSS_COMPILE	:= /usr/local/arm/4.3.1-eabi-armv6/usr/bin/arm-linux-
-#CROSS_COMPILE	:= $(shell if [ -f .cross_compile ]; then \
+# CROSS_COMPILE	:= /usr/local/arm/4.3.1-eabi-armv6/usr/bin/arm-linux-
+CROSS_COMPILE	:= $(shell if [ -f .cross_compile ]; then \
 					cat .cross_compile; \
 					fi)
 
