diff -NurbB a/os/linux/usb_main_dev.c b/os/linux/usb_main_dev.c
--- a/os/linux/usb_main_dev.c	2010-04-12 10:02:20.000000000 +0200
+++ b/os/linux/usb_main_dev.c	2010-05-27 19:25:45.000000000 +0200
@@ -326,7 +326,7 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,15)
 	.owner = THIS_MODULE,
 #endif	
-	.name="rt2870",
+	.name="rt3070",
 	.probe=rtusb_probe,
 	.disconnect=rtusb_disconnect,
 	.id_table=rtusb_dev_id,
@@ -393,7 +393,7 @@
 // Init driver module
 INT __init rtusb_init(void)
 {
-	printk("rtusb init --->\n");   
+	printk("rtusb (2.3.0.2) [SamyGO] init --->\n");   
 	return usb_register(&rtusb_driver);
 }
 
@@ -401,7 +401,7 @@
 VOID __exit rtusb_exit(void)
 {
 	usb_deregister(&rtusb_driver);	
-	printk("<--- rtusb exit\n");
+	printk("<--- rtusb (2.3.0.2) [SamyGO] exit\n");
 }
 
 module_init(rtusb_init);
