diff -NurbB djmount-0.71.orig/djmount/file_buffer.c djmount-0.71/djmount/file_buffer.c
--- djmount-0.71.orig/djmount/file_buffer.c	2006-08-27 22:12:20.000000000 +0200
+++ djmount-0.71/djmount/file_buffer.c	2010-01-24 19:56:12.000000000 +0100
@@ -168,15 +168,15 @@
 		 * UpnpReadHttpGet ...) has strange prototypes for length 
 		 * and ranges : "int" is not sufficient for large files !
 		 */
-		if (offset > FILE_BUFFER_MAX_CONTENT_LENGTH ||
-		    offset > FILE_BUFFER_MAX_CONTENT_LENGTH - size) {
-			Log_Printf (LOG_ERROR, 
-				    "GetHttp url '%s' overflowed "
-				    "size %" PRIdMAX " or offset %" PRIdMAX,
-				    file->url, (intmax_t) size, 
-				    (intmax_t) offset);
-			return -EOVERFLOW; // ---------->
-		}
+//		if (offset > FILE_BUFFER_MAX_CONTENT_LENGTH ||
+//		    offset > FILE_BUFFER_MAX_CONTENT_LENGTH - size) {
+//			Log_Printf (LOG_ERROR, 
+//				    "GetHttp url '%s' overflowed "
+//				    "size %" PRIdMAX " or offset %" PRIdMAX,
+//				    file->url, (intmax_t) size, 
+//				    (intmax_t) offset);
+//			return -EOVERFLOW; // ---------->
+//		}
 
 		// Adjust request to file size, if known
 		if (file->file_size >= 0) {
