diff -ur old/ipkg-build new/ipkg-build
--- old/ipkg-build	2010-04-22 07:31:36.000000000 +0200
+++ new/ipkg-build	2010-04-22 07:32:31.000000000 +0200
@@ -118,8 +118,8 @@
 	disallowed_filename=`disallowed_field Filename`
 	[ "$?" -ne 0 ] && PKG_ERROR=1
 
-	if echo $pkg | grep '[^a-z0-9.+-]'; then
-		echo "*** Error: Package name $name contains illegal characters, (other than [a-z0-9.+-])" >&2
+	if echo $pkg | grep '[^a-zA-Z0-9.+-]'; then
+		echo "*** Error: Package name $name contains illegal characters, (other than [a-zA-Z0-9.+-])" >&2
 		PKG_ERROR=1;
 	fi
 
