diff -ur old/opkg-build new/opkg-build
--- old/opkg-build	2010-05-08 16:38:31.000000000 +0200
+++ new/opkg-build	2010-05-08 16:40:56.000000000 +0200
@@ -118,8 +118,8 @@
 	disallowed_filename=`disallowed_field Filename`
 	[ "$?" -ne 0 ] && PKG_ERROR=1
 
-	if echo $pkg | grep '[^a-z0-9.+-]'; then
-		echo "*** Error: Package name $name contains illegal characters, (other than [a-z0-9.+-])" >&2
+	if echo $pkg | grep '[^a-zA-Z0-9.+-]'; then
+		echo "*** Error: Package name $name contains illegal characters, (other than [a-zA-Z0-9.+-])" >&2
 		PKG_ERROR=1;
 	fi
 
diff -ur old/opkg-buildpackage new/opkg-buildpackage
--- old/opkg-buildpackage	2010-05-08 16:38:31.000000000 +0200
+++ new/opkg-buildpackage	2010-05-08 16:41:22.000000000 +0200
@@ -65,8 +65,8 @@
 	required_field Maintainer >/dev/null
 	required_field Description >/dev/null
 
-	if echo $pkg | grep '[^a-z0-9.+-]'; then
-		echo "opkg-build: Error: Package name $name contains illegal characters, (other than [a-z0-9.+-])"
+	if echo $pkg | grep '[^a-zA-Z0-9.+-]'; then
+		echo "opkg-build: Error: Package name $name contains illegal characters, (other than [a-zA-Z0-9.+-])"
 		PKG_ERROR=1;
 	fi
 
